src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -0,0 +1,688 @@
+package org.folio.rest.impl;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static me.escoffier.vertx.completablefuture.VertxCompletableFuture.allOf;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.folio.orders.utils.ErrorCodes.*;
+import static org.folio.orders.utils.HelperUtils.collectResultsOnSuccess;
+import static org.folio.orders.utils.HelperUtils.encodeQuery;
+import static org.folio.orders.utils.HelperUtils.handleGetRequest;
+import static org.folio.orders.utils.HelperUtils.handlePutRequest;
+import static org.folio.orders.utils.ResourcePathResolver.PIECES;
+import static org.folio.orders.utils.ResourcePathResolver.PO_LINES;
+import static org.folio.orders.utils.ResourcePathResolver.resourceByIdPath;
+import static org.folio.orders.utils.ResourcePathResolver.resourcesPath;
+import static org.folio.rest.jaxrs.model.PoLine.ReceiptStatus.FULLY_RECEIVED;
+import static org.folio.rest.jaxrs.model.PoLine.ReceiptStatus.PARTIALLY_RECEIVED;
+import static org.folio.rest.jaxrs.model.PoLine.ReceiptStatus.AWAITING_RECEIPT;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import one.util.streamex.EntryStream;
+import one.util.streamex.StreamEx;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.orders.utils.HelperUtils;
+import org.folio.rest.acq.model.Piece;
+import org.folio.rest.acq.model.PieceCollection;
+import org.folio.rest.jaxrs.model.*;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.PoLine.ReceiptStatus;
+import org.folio.rest.acq.model.Piece.ReceivingStatus;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+public class CheckinReceivePiecesHelper<T> extends AbstractHelper {
+
+  static final int MAX_IDS_FOR_GET_RQ = 15;
+  private static final String PIECES_WITH_QUERY_ENDPOINT = resourcesPath(PIECES) + "?limit=%d&lang=%s&query=%s";
+  private static final String PIECES_BY_POL_ID_AND_STATUS_QUERY = "poLineId==%s and receivingStatus==%s";
+  Map<String, Map<String, T>> piecesByLineId;
+  private final InventoryHelper inventoryHelper;
+  Map<String, Map<String, Error>> processingErrors;
+  private final PurchaseOrderLineHelper poLineHelper;
+  private boolean isReceiving = false;
+  private boolean isCheckIn = false;
+
+  CheckinReceivePiecesHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders, Context ctx, String lang,
+      Map<String, Map<String, T>> piecesByLineId, Map<String, Map<String, Error>> processingErrors) {
+    super(httpClient, okapiHeaders, ctx, lang);
+    this.piecesByLineId = piecesByLineId;
+    inventoryHelper = new InventoryHelper(httpClient, okapiHeaders, ctx, lang);
+    this.processingErrors = processingErrors;
+    poLineHelper = new PurchaseOrderLineHelper(httpClient, okapiHeaders, ctx, lang);
+    isReceiving = StreamEx.ofValues(piecesByLineId)
```

Comment:
Probably it makes sense to make this class abstract and move specific logic to `CheckinHelper` and `ReceivingHelper` classes?

Reply:
correct, moved specific code into their own classes